### PR TITLE
Enable generatePackageTypings for some projects

### PR DESCRIPTION
**Pull request by pgonzal**
This PR fixes a few more PackageTypesGenerator edge cases, enables support in for this feature in GCB, and enables it for some preliminary projects.  Specific fixes:

- The \@packagedocumentation comment is now passed through to the emitted d.ts file
- Fixed a bug where certain definitions were incorrectly classified as ambient

### CODE REVIEW

File: common/changes/@microsoft/api-extractor/pgonzal-dts-generator-5_2018-01-12-01-56.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/api-extractor",
+      "comment": "Add some incremental improvements for the experimental PackageTypingsGenerator feature",
```
Comment (nickpape-msft) [inline common/changes/@microsoft/api-extractor/pgonzal-dts-generator-5_2018-01-12-01-56.json]: >some [](start = 22, length = 4)

do you want to elaborate?

File: apps/api-extractor/src/test/TypeScriptHelpers.test.ts [hunk 2]
```diff
@@ -122,4 +122,25 @@ describe('TypeScriptHelpers tests', () => {
 
   });
 
+  describe('formatJSDocContent()', () => {
+
+    const testCases: ITestCase[] = [
+      { // 0
+        input: 'this is\na test\n',
+        output: '/**\n * this is\n * a test\n */\n'
```
Comment (nickpape-msft) [inline apps/api-extractor/src/test/TypeScriptHelpers.test.ts]: is this covering the entire surface area of the function?

File: apps/api-extractor/src/TypeScriptHelpers.ts [hunk 3]
```diff
@@ -199,4 +199,35 @@ export default class TypeScriptHelpers {
...
     return lines.join('\n');
   }
+
+  /**
+   * Returns a JSDoc comment containing the provided content.
+   * @remarks
+   * This is the inverse of the extractJSDocContent() operation.
+   *
+   * WARNING: This function assumes that the content does not include strings
+   * such as "*\/" which would prematurely terminate the comment.
```
Comment (nickpape-msft) [inline apps/api-extractor/src/TypeScriptHelpers.ts]: should you add a test for this?

Comment (nickpape-msft) [inline apps/api-extractor/src/TypeScriptHelpers.ts]: >*\/ [](start = 14, length = 3)

handle this

File: common/changes/@microsoft/api-extractor/pgonzal-dts-generator-5_2018-01-12-01-56.json (hunk 1 above)
Comment (pgonzal) [reply common/changes/@microsoft/api-extractor/pgonzal-dts-generator-5_2018-01-12-01-56.json]: This feature is unreleased

---
In reply to: [161125673](https://github.com/Microsoft/web-build-tools/pull/483#discussion_r161125673) [](ancestors = 161125673)

**Pull request merged**